[DOCS] Note the cluster settings API can override `elasticsearch.yml`

jrodewig opened the pull request:
### Preview
https://elasticsearch_61394.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/settings.html#cluster-setting-types

jrodewig committed:
[DOCS] Note the cluster settings API can override `elasticsearch.yml`

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Settings)

elasticmachine commented:
Pinging @elastic/es-docs (>docs)

lcawl commented on docs/reference/setup/configuration.asciidoc:
```diff
@@ -98,7 +98,8 @@ You can configure and update dynamic settings on a running cluster using the
 <<cluster-update-settings,cluster update settings API>>.
```
> You can configure and update dynamic settings...

It's covered in the API, but is it also worth highlighting here that dynamic settings can be either persistent (apply across restarts) or transient (don’t survive a full cluster restart)?  That's something you don't get with static settings.

lcawl commented on docs/reference/setup/configuration.asciidoc:
```diff
@@ -98,7 +98,8 @@ You can configure and update dynamic settings on a running cluster using the
 +
 You can also configure dynamic settings locally on an unstarted or shut down
-node using `elasticsearch.yml`.
+node using `elasticsearch.yml`. However, any changes made using the cluster
+update settings API will override this file.
```
Not sure if it's more or less confusing, but here's another option:

```suggestion
node using `elasticsearch.yml`. However, the values shown in the <<cluster-get-settings,cluster get settings API>> 
override the values in the `elasticsearch.yml` configuration file.

jrodewig committed:
Update docs/reference/setup/configuration.asciidoc

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

jrodewig committed:
Incorporate feedback

jrodewig commented:
Thanks for the feedback @lcawl. I've updated the PR to cover persistent and transient settings. I also simplified the order of precedence so its clear it applies for resets or setting configurations using multiple methods.

jrodewig committed:
minor fixes

jrodewig committed:
clarify

jrodewig committed:
add example

jrodewig committed:
readd

jrodewig committed:
reword

lcawl reviewed (approved):
So much clearer now, thanks!
